src/porepy/numerics/fem/rt0.py

```diff
@@ -94,27 +94,37 @@ def discretize(self, g, data):
         dataIJ = np.empty(size)
         idx = 0
 
-        nodes, _, _ = sps.find(g.face_nodes)
-
         size_HB = g.dim * (g.dim + 1)
         HB = np.zeros((size_HB, size_HB))
         for it in np.arange(0, size_HB, g.dim):
             HB += np.diagflat(np.ones(size_HB - it), it)
         HB += HB.T
         HB /= g.dim * g.dim * (g.dim + 1) * (g.dim + 2)
 
+        # define the function to compute the inverse of the permeability matrix
+        if g.dim == 1:
+            inv_matrix = self._inv_matrix_1d
+        elif g.dim == 2:
+            inv_matrix = self._inv_matrix_2d
+        elif g.dim == 3:
+            inv_matrix = self._inv_matrix_3d
+
+        # compute the oppiside node per face
+        self._compute_cell_face_to_opposide_node(g, data)
+        cell_face_to_opposide_node = data[self.cell_face_to_opposide_node]
+
         for c in np.arange(g.num_cells):
```

Comment:
I guess if you want even more speedup, these loops should be possible to vectorize? Of cause not a must for this pull request

Reply:
that would be awesome, no idea how to do it :(

Reply:
Have you tried to numba it? That will go really fast, but it has a startup cost that often is a second or so. Or you can try cython.

See numercis.fv.fvutils.invert_diagonal_blocks() and associated functions for examples of both.